Lost serial connection

Hello,

Since zigpy 0.47.0 and zigpy-deconz 0.17.0 I got these error : 

```
[2022-06-22 09:06:22][ERROR] : Lost serial connection: device reports readiness to read but returned no data (device disconnected or multiple access on port?)
[2022-06-22 09:06:22][WARNING] : Serial '/dev/ttyACM0' connection lost unexpectedly: device reports readiness to read but returned no data (device disconnected or multiple access on port?)
[2022-06-22 09:06:22][DEBUG] : Reconnecting '/dev/ttyACM0' serial port
[2022-06-22 09:06:22][DEBUG] : Couldn't re-open '/dev/ttyACM0' serial port, retrying in 2s: [Errno 2] could not open port /dev/ttyACM0: [Errno 2] No such file or directory: '/dev/ttyACM0'
[2022-06-22 09:06:24][DEBUG] : Reconnecting '/dev/ttyACM0' serial port
[2022-06-22 09:06:24][DEBUG] : Connection made
[2022-06-22 09:06:24][DEBUG] : Reconnected '/dev/ttyACM0' serial port after 2 attempts
[2022-06-22 09:06:27][ERROR] : Lost serial connection: device reports readiness to read but returned no data (device disconnected or multiple access on port?)
[2022-06-22 09:06:27][WARNING] : Serial '/dev/ttyACM0' connection lost unexpectedly: device reports readiness to read but returned no data (device disconnected or multiple access on port?)
[2022-06-22 09:06:27][DEBUG] : Reconnecting '/dev/ttyACM0' serial port
[2022-06-22 09:06:27][DEBUG] : Couldn't re-open '/dev/ttyACM0' serial port, retrying in 2s: [Errno 2] could not open port /dev/ttyACM0: [Errno 2] No such file or directory: '/dev/ttyACM0'
[2022-06-22 09:06:29][DEBUG] : Reconnecting '/dev/ttyACM0' serial port
[2022-06-22 09:06:29][DEBUG] : Connection made
[2022-06-22 09:06:29][DEBUG] : Reconnected '/dev/ttyACM0' serial port after 2 attempts
```

I know this error is normally hardware error but before 0.47 no issue, I test to downgrad to 0.46 and zigpy-deconz 0.16 and error disapear.

For information I'am on firmware 0x26720700.

Thank in advance for your time

--- a/tests/test_application.py
+++ b/tests/test_application.py
@@ -252,15 +252,12 @@ async def test_start_network(app, proto_ver, nwk_state, error):
     with patch.object(application.DeconzDevice, "initialize", AsyncMock()):
         await app.start_network()
         assert app.load_network_info.await_count == 1
+        assert app._change_network_state.await_count == 1
 
-        if nwk_state != deconz_api.NetworkState.CONNECTED:
-            assert app._change_network_state.await_count == 1
-            assert (
-                app._change_network_state.await_args_list[0][0][0]
-                == deconz_api.NetworkState.CONNECTED
-            )
-        else:
-            assert app._change_network_state.await_count == 0
+        assert (
+            app._change_network_state.await_args_list[0][0][0]
+            == deconz_api.NetworkState.CONNECTED
+        )
 
         if proto_ver >= application.PROTO_VER_NEIGBOURS:
             assert app.restore_neighbours.await_count == 1

--- a/zigpy_deconz/zigbee/application.py
+++ b/zigpy_deconz/zigbee/application.py
@@ -80,8 +80,9 @@ class ControllerApplication(zigpy.application.ControllerApplication):
                 await self._api.write_parameter(
                     NetworkParameter.watchdog_ttl, self._config[CONF_WATCHDOG_TTL]
                 )
-            except (asyncio.TimeoutError, zigpy.exceptions.ZigbeeException):
-                LOGGER.warning("No watchdog response")
+            except Exception as e:
+                LOGGER.warning("Failed to reset watchdog", exc_info=e)
+
             await asyncio.sleep(self._config[CONF_WATCHDOG_TTL] * 0.75)
 
     async def connect(self):
@@ -95,15 +96,7 @@ class ControllerApplication(zigpy.application.ControllerApplication):
         if self._reset_watchdog_task is not None:
             self._reset_watchdog_task.cancel()
 
-        if self._api is None:
-            return
-
-        try:
-            if self._api.protocol_version >= PROTO_VER_WATCHDOG:
-                await self._api.write_parameter(NetworkParameter.watchdog_ttl, 1)
-        except zigpy_deconz.exception.CommandError:
-            pass
-        finally:
+        if self._api is not None:
             self._api.close()
 
     async def permit_with_key(self, node: t.EUI64, code: bytes, time_s=60):
@@ -113,13 +106,10 @@ class ControllerApplication(zigpy.application.ControllerApplication):
         await self.register_endpoints()
         await self.load_network_info(load_devices=False)
 
-        device_state, _, _ = await self._api.device_state()
-
-        if device_state.network_state != NetworkState.CONNECTED:
-            try:
-                await self._change_network_state(NetworkState.CONNECTED)
-            except asyncio.TimeoutError as e:
-                raise FormationFailure() from e
+        try:
+            await self._change_network_state(NetworkState.CONNECTED)
+        except asyncio.TimeoutError as e:
+            raise FormationFailure() from e
 
         coordinator = await DeconzDevice.new(
             self,

